### PR TITLE
Prac 05 feedback

### PR DESCRIPTION
**Pull request by nora-zyy**
Can you comment for me, please.
@a-rosa
@Lijiayi2
@Adrianto Hartono

### CODE REVIEW

**Comment by nora-zyy** [comment]
@AdriantoHartono
@a-rosa
@Lijiayi2

File: prac_05/emails.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    name = input(f"Is your name {username.title()}? (Y/n) ")
+    if not name or name.lower() == "y":
+        email_name[email] = username
+    elif name.lower() == "n":
+        name = input("Name: ")
+        email_name[email] = name
+
+for email, name in email_name.items():
+    print(f"{name.title()} ({email})")
+
```
Comment (Lijiayi2) [inline prac_05/emails.py:21]: Good code

File: prac_05/word_occurrences.py
Comment (Lijiayi2) [inline prac_05/word_occurrences.py:1]: Good code

File: prac_05/state_names.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+while state_code != "".upper():
+    while True:
+        try:
+            state = input("Enter short state:").upper()
+            if state.lower() == 'quit':
+                break
+            capital = CODE_TO_NAME[state.upper()]
+            print(f"{state.upper()} is {capital}")
+        except KeyError:
+            print("Enter short state: ")
```
Comment (Lijiayi2) [inline prac_05/state_names.py:22]: I think that you should add .upper() in this line

File: prac_05/emails.py (hunk 1 above)
Comment (AdriantoHartono) [reply prac_05/emails.py:21]: Good code, it can function as expected 🙏

File: prac_05/state_names.py (hunk 2 above)
Comment (AdriantoHartono) [reply prac_05/state_names.py:22]: The code already good, just don't forget to put the .upper() behind the input to make sure all the input can be read

File: prac_05/word_occurrences.py
Comment (AdriantoHartono) [reply prac_05/word_occurrences.py:1]: The code has been well organized with all the functions 👍

**Review by AdriantoHartono — COMMENTED** [review]
So far, all the code already well organized. But some of the code just missed a little part 👌

**Comment by nora-zyy** [comment]
@ a-rosa

File: prac_05/emails.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+""""
+estimate: 30 mins
+actual: 40 mins
+"""
+email_name = {}
+
```
Comment (a-rosa) [inline prac_05/emails.py:6]: You could just use while email is not empty instead and there would no be any need to have break

File: prac_05/emails.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+while True:
+    email = input("Email: ")
+    if not email:
+        break
+    username = email.split('@')[0]
```
Comment (a-rosa) [inline prac_05/emails.py:11]: You have forgotten that emails might have "." in between names such as lindsay.ward needs to be printed as lindsay ward not lindsay.ward

File: prac_05/hex_colours.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    "Absolute Zero": "#0048ba","AliceBlue": "##f0f8ff",
+    "Baby Blue": "#89cff0","Baby Pink": "#f4c2c2",
+    "Cadet Grey": "#91a3b0","CadetBlue1": "#98f5ff",
+    "Dark Sky Blue": "#8cbed6","DarkOrchiz": "#9932cc",
+    "Earth Yellow": "#e1a95f","Electric Blue": "#7df9ff",
+}
+print(COLORS)
+
+while True:
+    color_name = input('Choice color:').strip().upper()
```
Comment (a-rosa) [inline prac_05/hex_colours.py:11]: You don't need to strip it

File: prac_05/hex_colours.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    if not color_name:
+        break
```
Comment (a-rosa) [inline prac_05/hex_colours.py:13]: Instead of break, it would be better to print an error message

File: prac_05/state_names.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+State names in a dictionary
+File needs reformatting
+"""
+
+# TODO: Reformat this file so the dictionary code follows PEP 8 convention
+CODE_TO_NAME = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
+            "ACT": "Australian Capital Territory", "VIC": "Victoria", "TAS": "Tasmania"}
+print(CODE_TO_NAME)
+
+state_code = input("Enter short state: ")
```
Comment (a-rosa) [inline prac_05/state_names.py:12]: missing .upper in the end

File: prac_05/state_names.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+while state_code != "".upper():
+    while True:
```
Comment (a-rosa) [inline prac_05/state_names.py:14]: You don't need the while true and asking another time, instead immediately go to if else

File: prac_05/wimbledon.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    print("Wimbledon Champions:")
+    for champion, count in champions.items():
+        print(f"{champion} {count}")
+
+def print_countries(countries):
+    print(f"These {len(countries)} countries have won Wimbledon:")
+    print(", ".join(countries))
+
+
+main()
```
Comment (a-rosa) [inline prac_05/wimbledon.py:50]: Forgotten docstring in all functions

File: prac_05/word_occurrences.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+for word in word_list:
+    if len(word) > max_width:
+        max_width = len(word)
+    if word in word_count:
+        word_count[word] += 1
+    else:
+        word_count[word] = 1
+
+for word, count in word_count.items():
+    print(f"{word:{max_width}} : {count}")
```
Comment (a-rosa) [inline prac_05/word_occurrences.py:15]: Forgotten the estimated and actual on top of the program

**Pull request merged**